main.py

```diff
@@ -55,10 +55,27 @@ def write_cache(live_cache: dict):
     with open("roles.json", "w") as f:
         json.dump(live_cache, f, indent=2)
 
+#Below is the code for sending ZOOM LINK by Abir
+async def zoom_link():    
+    now = datetime.datetime.weekday(datetime.datetime.now())    #Assigned now that return a number between 0 - 6 depending on the current time. Current time given as argument inside parenthesis.
+    if now == 2:  #0 == monday, 1 == tues, 2 == for wed ... sunday == 6.  
+        #Below, the code is for sending a message at a specific time  
+        time_now = datetime.datetime.now()   #Assigning timenow with current time as value
+        then = time_now.replace(hour=10,minute=00)   #Assigning then with specific time we want the message to be executed
+        wait_time = (then-time_now).total_seconds() #(then - timenow) will return amount of time(in seconds) for the program to wait till reaching the given time in order to execute the message 
+        await asyncio.sleep(wait_time) #asyncio.sleep will delay the program for the amount of time assigned in var wait_time
+
+        #Displaying message to users 
+        channel = bot.get_channel(1046243880057712686) #The ID for the channel we want the message to display
+        await channel.send("Hey Fellow Programmers! Join Today's Club meeting by clicking the Zoom link below:")
+        await channel.send("Link: https://bmcc-cuny.zoom.us/j/89372621235?pwd=UU1CTXphUm5qSjRMTk5oNFZsVjdwdz09")
+
+      
 
 @bot.event
 async def on_ready():
     print(f'Bot is logged in as {bot.user}')
+    await zoom_link()
```

Comment:
This function is being called in `on_ready()`, a default function that runs once every time the bot starts up. In other words, this function zoom_link is called once when the bot starts up and never again. 

<s>I think the solution may be to create some sort of loop using dateTime that runs the function once per week. 

Here's a couple resources to look at:
https://pypi.org/project/schedule/
https://youtu.be/ovT9GQ-0mlU</s>

Wait never mind, I did some digging and found this:
https://discordpy.readthedocs.io/en/stable/ext/tasks/index.html?highlight=loop#discord.ext.tasks.Loop

It is a decorator which is like these: `@bot.command()`, `@commands.has_permissions(administrator=True)`, etc...

You can "decorate" your function and then it will automatically run once per week!
Read the documentation on how to do it. I actually found this from someone else's discord bot here:
https://github.com/HunterSandlin/DiscordDailyMessageBot/blob/master/DailyMessageBot.py

Reply:
Fixed! I made two changes. Firstly, added tasks.loop(hours=24) on top of the zoom_link function. This creates a loop that will iterate the zoom_link function every 24 hours when program is running. 
Another change is - under the on_ready function, I added zoom_link.start(). This is important because this line is what starts the loop. Therefore, now the zoom_link function will automatically run once per week.